WooCommerce/WooCommerceTests/POS/ViewModels/ItemListViewModelTests.swift

```diff
@@ -211,21 +224,38 @@ final class ItemListViewModelTests: XCTestCase {
         // When
         await sut.populatePointOfSaleItems()
 
+        sut.$shouldShowHeaderBanner
+            .dropFirst()
+            .removeDuplicates()
+            .sink { value in
+            XCTAssertEqual(value, false)
+            expectation.fulfill()
+        }
+        .store(in: &cancellables)
+
         // Then
         XCTAssertEqual(sut.state, .error(expectedError))
-        XCTAssertEqual(sut.shouldShowHeaderBanner, false)
     }
 
     func test_shouldShowHeaderBanner_when_itemListViewModel_loaded_then_returns_true() async {
         // Given
+        let expectation = XCTestExpectation(description: "Banner should be shown once products are loaded")
         let expectedItems = Self.makeItems()
 
         // When
         await sut.populatePointOfSaleItems()
 
+        sut.$shouldShowHeaderBanner
```

Comment:
I would guess that fulfilling expectations should be enough to avoid weird CI issues. 

Now when we don't wait for expectations to be fulfilled, our test can succeed when `XCTAssertEqual(sut.state, .loaded(expectedItems))` is called. Although we want to test whether `shouldShowHeaderBanner` is true or false.